ceph/basic-pynfs-cthon.sh

```diff
@@ -0,0 +1,279 @@
+            exit 1
+        fi
+    fi
+
+    sleep 5
+done
+
+# view the osd's
+ceph osd tree
+# Create a cephfs volume
+ceph fs volume create cephfs
+
+# Create subvolumegroup
+ceph fs subvolumegroup create cephfs ganeshagroup
+
+# Create subvolume
+ceph fs subvolume create cephfs nfs_subvol --group_name ganeshagroup --namespace-isolated
+
+# Get subvolume path
+CEPHFS_NAME="cephfs"
+SUBVOLUME_NAME="nfs_subvol"
+GROUP_NAME="ganeshagroup"
+SUBVOL_PATH=$(ceph fs subvolume getpath "$CEPHFS_NAME" "$SUBVOLUME_NAME" --group_name "$GROUP_NAME" 2>/dev/null)
+
+# Verify path was obtained
+if [ -z "$SUBVOL_PATH" ]; then
+    echo "ERROR: Failed to get subvolume path."
+    exit 1
+fi
+echo "Subvolume path: $SUBVOL_PATH"
+
+# create ganesha.conf file
+echo "NFS_CORE_PARAM {
+    Enable_NLM = false;
+    Enable_RQUOTA = false;
+    Protocols = 4;
+}
+
+EXPORT_DEFAULTS {
+    Access_Type = RW;
+}
+EXPORT {
+    Export_ID = 101;
+    Path = \"$SUBVOL_PATH\";
+    Pseudo = \"/nfs/cephfs\";
+    Protocols = 4;
+    Transports = TCP;
+    Access_Type = RW;
+    Squash = None;
+    FSAL {
+        Name = \"CEPH\";
+    }
+}" > /etc/ganesha/ganesha.conf
+
+# View the ganesha conf file
+cat /etc/ganesha/ganesha.conf
+
+mkdir -p /var/run/ganesha
+chmod 755 /var/run/ganesha
+chown root:root /var/run/ganesha
+
+# Creating backend recovery dir for nfs ganesha
+mkdir -p /var/lib/nfs/ganesha
+chmod 755 /var/lib/nfs/ganesha
+chown root:root /var/lib/nfs/ganesha
+
+ganesha.nfsd -f /etc/ganesha/ganesha.conf -L /var/log/ganesha.log
+if pgrep ganesha >/dev/null; then
+        echo "[OK] Service ganesha is running"
+        echo $(pgrep ganesha)
+    else
+        echo "[ERROR] Service ganesha is NOT running" >&2
+        exit 1
+fi
+
+# Run Cthon post successful cluster creation
+
+# install build and runtime dependencies
+yum -y install git gcc nfs-utils time make
+
+#!/bin/sh
+#
+# Environment variables used:
+#  - SERVER: hostname or IP-address of the NFS-server
+#  - EXPORT: NFS-export to test (should start with "/")
+
+echo "Client Script"
+
+# enable some more output
+set -x
+
+[ -n "${SERVER}" ]
+[ -n "${EXPORT}" ]
+[ -n "${TEST_PARAMETERS}" ]
+
+SERVER='$(hostname -I | awk '{print $1}')'
+EXPORT='/nfs/cephfs'
+
+
+# install build and runtime dependencies
+dnf -y install git gcc nfs-utils redhat-rpm-config krb5-devel python3-devel python3-gssapi python3-ply
+
+rm -rf /root/pynfs && git clone git://linux-nfs.org/~bfields/pynfs.git
+
+cd /root/pynfs && yes | python3 setup.py build > /tmp/output_tempfile.txt
+echo $?
+
+LOG_FILE40="/tmp/pynfs"$(date +%s)".log"
+cd /root/pynfs/nfs4.0
+./testserver.py ${SERVER}:${EXPORT} --verbose --maketree --showomit --rundeps all ganesha ${TEST_PARAMETERS} >> "${LOG_FILE40}"
+RETURN_CODE40=$?
+
+echo "pynfs 4.0 test output:"
+cat $LOG_FILE40
+
+LOG_FILE41="/tmp/pynfs"$(date +%s)".log"
+cd /root/pynfs/nfs4.1
+./testserver.py ${SERVER}:${EXPORT} all ganesha --verbose --maketree --showomit --rundeps >> "${LOG_FILE41}"
```

Comment:
Likewise, we can switch to using `--outfile`.

I strongly believe, we should use `build_scripts/common/pynfs/client.sh` instead of adding the automation here. This way we can reuse exisitng code.